changelog/template_alias_matches_basic_types.dd

```diff
@@ -0,0 +1,22 @@
+Template alias parameters now match basic types as a conversion.
+
+With this release, templates with alias parameters can be instantiated with basic types,
+such as `int` or `void function()`.
+
+---
+template Example(alias A) { alias Example = A[]; }
+alias IntArray = Example!int;
+
+template MatchType(T) { enum MatchType = "type"; }
+template MatchType(alias A) { enum MatchType = "alias"; }
+
+// alias matches, but type matches take priority.
+static assert (MatchType!int == "type");
+---
+
+Templates with alias parameters already matched named types, lambdas, and even value literals. Templates with
+variadic parameters already matched basic types. This led to strange replacements for `alias` such as
+`template Template(T...) if (T.length == 1)`. Those replacements are no longer necessary.
+
+As a consequence, in some cases templates will match that did not used to. As this may break existing code,
```

Comment:
Extra points for showing what cases break, and _how to fix it_. :)

Reply:
Better?

Reply:
thanks. Perhaps add explicitly that the code would break for delegates.
I guess things can break in many different ways. Regardless, the example shows the type of fix that is required.